frame/executive/src/lib.rs

```diff
@@ -222,24 +224,39 @@ where
 	OriginOf<Block::Extrinsic, Context>: From<Option<System::AccountId>>,
 	UnsignedValidator: ValidateUnsigned<Call = CallOf<Block::Extrinsic, Context>>,
 {
-	/// Execute all `OnRuntimeUpgrade` of this runtime, and return the aggregate weight.
-	pub fn execute_on_runtime_upgrade() -> frame_support::weights::Weight {
-		<(COnRuntimeUpgrade, AllPalletsWithSystem) as OnRuntimeUpgrade>::on_runtime_upgrade()
-	}
-
-	/// Execute given block, but don't do any of the `final_checks`.
+	/// Execute given block, but don't as strict is the normal block execution.
 	///
-	/// Should only be used for testing.
-	#[cfg(feature = "try-runtime")]
-	pub fn execute_block_no_check(block: Block) -> frame_support::weights::Weight {
+	/// Some consensus related checks such as the state root check can be switched off via
+	/// `try_state_root`. Some additional non-consensus checks can be additionally enabled via
+	/// `try_state`.
+	///
+	/// Should only be used for testing ONLY.
+	pub fn try_execute_block(
+		block: Block,
+		try_state_root: bool,
+		select: frame_try_runtime::TryStateSelect,
+	) -> Result<frame_support::weights::Weight, &'static str> {
+		use frame_support::traits::TryState;
+
 		Self::initialize_block(block.header());
 		Self::initial_checks(&block);
 
 		let (header, extrinsics) = block.deconstruct();
 
 		Self::execute_extrinsics_with_book_keeping(extrinsics, *header.number());
 
-		// do some of the checks that would normally happen in `final_checks`, but definitely skip
+		// run the try-state checks of all pallets.
+		<AllPalletsWithSystem as TryState<System::BlockNumber>>::try_state(
+			*header.number(),
+			select,
+		)
+		.map_err(|e| {
+			sp_runtime::print("failure:");
+			sp_runtime::print(e);
+			e
+		})?;
+
+		// do some of the checks that would normally happen in `final_checks`, but perhaps skip
 		// the state root check.
 		{
 			let new_header = <frame_system::Pallet<System>>::finalize();
```

Comment:
Assuming you are running with a different runtime, this could be easily something that is different :see_no_evil: 

Maybe we should just disable all checks if wanted.

Reply:
I will leave that up to https://github.com/paritytech/substrate/pull/12048#issuecomment-1228427709 and @pmikolajczyk41